tests/test_reqparse.py

```diff
@@ -723,6 +723,39 @@ def test_strict_parsing_on_partial_hit(self):
         parser.add_argument('foo', type=int)
         self.assertRaises(exceptions.BadRequest, parser.parse_args, req, strict=True)
 
+    def test_trim_argument(self):
+        req = Request.from_values("/bubble?foo= 1 &bar=bees&n=22")
```

Comment:
The actual trim takes place before argument is cast into specified datatype. I added a hasattr safeguard, as it is also done before lower() call:
    if hasattr(value, "strip") and self.trim:
        value = value.strip()
    if hasattr(value, "lower") and not self.case_sensitive:                                                                   
            value = value.lower()